apprt/gtk: implement context menu

kareigu opened the pull request:
Implements a context menu for GTK:

## Without selection:
![image](https://github.com/user-attachments/assets/fa53bf7b-3b05-4fc5-be13-80732482fa73)

## With selection:
![image](https://github.com/user-attachments/assets/8917448d-04bf-455b-83f5-8be01d6656c6)

Tried to match behaviour in other GTK apps with the menu preferring to open down and right from the cursor, if possible.

## A couple notes:
Left a FIXME in place for disabling copy while nothing is selected. Currently I'm just setting the button action to `noop` which feels really hackish, but from trying multiple different approaches this was the best one by far. (Suggestions welcome).

Setting sensitivity to disabled on the menu item didn't seem to work from what I tried, while removing the menu item completely causes some nasty artifacts where the menu is displayed when it still hasn't finished doing layout again.

**Which brings me to the second note:**
Currently we're storing a reference to the same `PopoverMenu` and reusing it when we could be doing the same approach as with AppKit and just creating it on the fly as needed. 

This however causes some problems like the menu becoming a scrollable list for some reason (this I suspect to be just a skill issue on my end, since I can't find other GTK apps doing the same thing) and having this ugly relayout happening when you open the menu. 

*Nautilus creates a new `PopoverMenu` every time you right click on something*

https://github.com/user-attachments/assets/c49730ae-04eb-4a3f-9b07-b88b32ee7ef1

The video doesn't really show it well on anything other than the longer menus, but in person it's really noticeable.

kareigu committed:
apprt/gtk: implement context menu

Implements context menu for GTK with:
- copy
- paste
- split right
- split down
- terminal inspector

kareigu committed:
apprt/gtk: disable copy in context menu while without selection

Left a FIXME where the "Copy" button action is disabled.

Though very hackish this was the best way I found to do this currently.
Disable sensitivity on the button didn't do anything and trying to
remove the button altogether like on macOS, causes the menu to become
really buggy. Either by the context menu turning into a scrollable list
or by it becoming really janky and showing the user pre-update UI.

mitchellh reviewed (changes requested):
Thanks for doing this! Looks pretty good. 

Noted on your gotchas, I'll take a look but I'd be fine merging the logic as-is today and looking into improving it later too.

mitchellh committed:
apprt/gtk: get rid of forcing context menu for now

We have escapes (shift) so lets see how that goes and compare to some
other GTK apps first.

mitchellh committed:
apprt/gtk: some stylistic changes

kareigu committed:
apprt/gtk: don't dim surface when opening context menu
